esp/platform/espprotocol.cpp

```diff
@@ -28,21 +28,21 @@
 
 typedef IXslProcessor * (*getXslProcessor_func)();
 
-static atomic_t gActiveRequests;
+static std::atomic<unsigned> gActiveRequests;
```

Comment:
Concern here is that this will be used a metric (I have a prototype running that accepts a reference to the underlying metric value). We decided to use 64bit values, so my prototype uses std::atomic<__uint64> gActiveRequests;

Reply:
This is a conversion of what is there already.  I suspect your metric classes that use a value that is calculated elsewhere should be an example of a template class - it reports as an uin64, but can read from any value.